src/main/java/com/teamscale/buildbreaker/commandline/BuildBreaker.java

```diff
@@ -0,0 +1,428 @@
+            throw new InvalidParametersException(
+                    "Please specify at least one of --evaluate-findings or --evaluate-thresholds, otherwise no evaluation will take place.");
+        }
+        OkHttpClient okHttpClient = OkHttpClientUtils
+                .createClient(sslConnectionOptions.disableSslValidation, sslConnectionOptions.keyStorePath,
+                        sslConnectionOptions.keyStorePassword);
+        teamscaleClient = new TeamscaleClient(okHttpClient, teamscaleServerUrl, user, accessKey, project);
+        EvaluationResult aggregatedResult = new EvaluationResult();
+
+        try {
+            waitForAnalysisToFinish(determineBranchAndTimestamp());
+            if (thresholdEvalOptions.evaluateThresholds) {
+                aggregatedResult.addAll(evaluateMetrics());
+            }
+
+            if (findingEvalOptions.evaluateFindings) {
+                aggregatedResult.addAll(evaluateFindings());
+            }
+            return aggregatedResult.toStatusCode();
+        } catch (SSLHandshakeException e) {
+            handleSslConnectionFailure(e);
+        } catch (UnknownHostException e) {
+            fail("The host " + teamscaleServerUrl + " could not be resolved. Please ensure you have no typo and that" +
+                    " this host is reachable from this server. " + e.getMessage());
+        } catch (ConnectException e) {
+            fail("The URL " + teamscaleServerUrl + " refused a connection. Please ensure that you have no typo and that" +
+                    " this endpoint is reachable and not blocked by firewalls. " + e.getMessage());
+        } catch (HttpRedirectException e) {
+            fail("You provided an incorrect URL. The server responded with a redirect to " + "'" + e.getRedirectLocation() + "'." +
+                    " This may e.g. happen if you used HTTP instead of HTTPS." +
+                    " Please use the correct URL for Teamscale instead.");
+        } catch (HttpStatusCodeException e) {
+            handleHttpStatusCodeException(e);
+        } catch (CommitCouldNotBeResolvedException e) {
+            // We do not call fail here because we want to keep the old api of returning code -5
+            System.out.println("Could not resolve revision " + e.getRevision() +
+                    " to a valid commit known to Teamscale (no commits returned)");
+            return -5;
+        } catch (TooManyCommitsException e) {
+            fail("Could not resolve revision " + e.getRevision() +
+                    " to a valid commit known to Teamscale (too many commits returned): " + e.getCommitDescriptorsJson());
+        } catch (IOException e) {
+            fail("Encountered an error while communicating with Teamscale: ");
+        } finally {
+            // we must shut down OkHttp as otherwise it will leave threads running and
+            // prevent JVM shutdown
+            teamscaleClient.close();
```

Comment:
I wanted to use try with resources but we need the client in other methods and it felt cumbersome to pass it everywhere. So I opted to close the client manually.